MyModels/BaseModel.py

```diff
@@ -0,0 +1,46 @@
+class BaseModel:
+    def __init__(self):
+        self.list = []
+        self.id_index = {}
+        self.id_sequence_counter = 0
+        
+    def add(self, entry):
+        
+        self.list.append(entry)
+
+        userid = entry['userid'] # Applicable in Contact.py Only
+        if userid:
+            if self.userid_id.get(userid):
+                self.userid_id[userid].append(self.id_sequence_counter)
+            else:
+                self.userid_id[userid] = [self.id_sequence_counter]
+            
+        self.id_index[self.id_sequence_counter] = len(self.list) - 1
+        
+        self.id_sequence_counter += 1
+
+
+    def find_index(self,id_sequence_counter):
```

Comment:
????? What does `id_sequence_counter` have to do here?

Reply:
for the sake of readability! did it backfired?!